### ISSUE TITLE
VaultPKISecret altNames type conversion error

### ISSUE THREAD

**Issue by andrejvanderzee** [hashicorp/vault-secrets-operator#189]
**Describe the bug**

Specifying altNames for VaultPKISecret results in reconciliation error:

```
* Field validation failed: error converting input [test test.infra.svc test.infra.svc.cluster.local] for field "alt_names": '' expected type 'string', got unconvertible type '[]interface {}', value: '[test test.infra.svc test.infra.svc.cluster.local]'	{"type": "Warning", "object": {"kind":"VaultPKISecret","namespace":"infra","name":"testing","uid":"12fa9871-496e-4af5-a079-09b4b9a4f45b","apiVersion":"secrets.hashicorp.com/v1alpha1","resourceVersion":"101270849"}, "reason": "K8sClientError"}
2023-05-19T10:19:28Z	ERROR	Reconciler error	{"controller": "vaultpkisecret", "controllerGroup": "secrets.hashicorp.com", "controllerKind": "VaultPKISecret", "VaultPKISecret": {"name":"testing","namespace":"infra"}, "namespace": "infra", "name": "testing", "reconcileID": "9a06346a-bfd0-4175-8ae0-69c69b9bf88c", "error": "Error making API request.\n\nURL: PUT https://vault.perfana.cloud/v1/management/eks/issue/server\nCode: 400. Errors:\n\n* Field validation failed: error converting input [test test.infra.svc test.infra.svc.cluster.local] for field \"alt_names\": '' expected type 'string', got unconvertible type '[]interface {}', value: '[test test.infra.svc test.infra.svc.cluster.local]'"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:329
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:274
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:235
```

**To Reproduce**

Apply a CRD similar to this:

```
apiVersion: secrets.hashicorp.com/v1alpha1
kind: VaultPKISecret
metadata:
  name: testing
  namespace: infra
spec:
  vaultAuthRef: defaultaltNames
  namespace: infra
  mount: eks/pki
  name: server
  destination:
    create: true
    name: vault-secrets-operator-test
    type: kubernetes.io/tls
  commonName: test.example.com
  format: pem
  revoke: true
  clear: true
  expiryOffset: 15s
  ttl: "5m"
  altNames:
  - "test"
  - "test.infra.svc"
  - "test.infra.svc.cluster.local"
```

**Expected behavior**

TLS secret tobe created with altNames.

**Environment**

kubernetes: EKS v1.25.6
vault-secrets-operator: v0.1.0-beta

**Comment by benashz** [comment]
Thank you for reporting this issue @andrejvanderzee. We will take a look at it soon.

**Comment by benashz** [comment]
Hi @andrejvanderzee , this issue should be fixed with #130. It should go out in the upcoming beta release.

I am going to close this issue as a duplicate #115. If you continue to experience a similar issue please feel free to open a new GH issue.

Thanks,

Ben

**Comment by andrejvanderzee** [comment]
@benashz Thank you. Yes I can verify it works properly, as I switched to the main-branch instead of the beta-release.
